src/format.rs

```diff
@@ -0,0 +1,392 @@
+//                                                                                //
+//                                                                                //
+//                                                                                //
+//                                config.toml                                     //
+//                                                                                //
+//                                                                                //
+//                                                                                //
+////////////////////////////////////////////////////////////////////////////////////
+
+/// config.toml
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct ConfigFile {
+    /// This top-level key specifies the default criteria that cargo vet certify will use
+    /// when recording audits. If unspecified, this defaults to "safe-to-deploy".
+    #[serde(rename = "default-criteria")]
+    #[serde(default = "get_default_criteria")]
+    #[serde(skip_serializing_if = "is_default_criteria")]
+    pub default_criteria: String,
+    /// Remote audits.toml's that we trust and want to import.
+    pub imports: StableMap<String, RemoteImport>,
+    /// All of the "foreign" dependencies that we rely on but haven't audited yet.
+    /// Foreign dependencies are just "things on crates.io", everything else
+    /// (paths, git, etc) is assumed to be "under your control" and therefore implicitly trusted.
+    pub unaudited: StableMap<String, Vec<UnauditedDependency>>,
+    pub policy: PolicyTable,
+}
+
+impl ConfigFile {
+    pub fn validate(&self) -> Result<(), VetError> {
+        // TODO
+        Ok(())
+    }
+}
+
+pub static SAFE_TO_DEPLOY: &str = "safe-to-deploy";
+pub static SAFE_TO_RUN: &str = "safe-to-run";
+pub static DEFAULT_CRITERIA: &str = SAFE_TO_DEPLOY;
+
+pub fn get_default_criteria() -> String {
+    String::from(DEFAULT_CRITERIA)
+}
+fn is_default_criteria(val: &String) -> bool {
+    val == DEFAULT_CRITERIA
+}
+
+/// Policies that first-party (non-foreign) crates must pass.
+///
+/// This is basically the first-party equivalent of audits.toml, which is separated out
+/// because it's not supposed to be shared (or, doesn't really make sense to share,
+/// since first-party crates are defined by "not on crates.io").
+///
+/// Because first-party crates are implicitly trusted, really the only purpose of this
+/// table is to define the boundary between a first-party crates and third-party ones.
+/// More specifically, the criteria of the dependency edges between a first-party crate
+/// and its direct third-party dependencies.
+///
+/// If this sounds overwhelming, don't worry, everything defaults to "nothing special"
+/// and an empty PolicyTable basically just means "everything should satisfy the
+/// default criteria in audits.toml".
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct PolicyTable {
```

Comment:
Shouldn't this be PolicyEntry?

Reply:
Yes so I have this entire time been operating on the understanding that there is One PolicyTable Defining Defaults and not that it's `[policy.my_crate]` like with audits. I don't know if you changed this at one point, or if I have just been missing that fact the whole time, but the docs don't directly state that policy tables are keyed on a crate? It's just implied by the prose talking about "the" crate.

This a pre-existing error(?) though, this PR doesn't change that. Policy stuff will be cleaned up in the next round since policy and suggest are very fundamentally intertwined.